### PR TITLE
AP_Avoidance: Tidy construction of temporary objects on stack

### PR DESCRIPTION
**Pull request by peterbarker**
trying to reduce places that know the innards of AP_Location

Tested in SITL by creating a swarm, fiddling DEV_OPTION bit....

### CODE REVIEW

File: libraries/AP_Avoidance/AP_Avoidance.cpp [hunk 1]
```diff
@@ -593,14 +593,17 @@ void AP_Avoidance::handle_msg(const mavlink_message_t &msg)
...
-    loc.lng = packet.lon;
-    loc.alt = packet.alt / 10; // mm -> cm
-    loc.relative_alt = false;
-    Vector3f vel = Vector3f(packet.vx/100.0f, // cm to m
-                            packet.vy/100.0f,
-                            packet.vz/100.0f);
+    const Location loc {
+        packet.lat,
+        packet.lon,
+        int32_t(packet.alt * 0.01),  // mm -> cm
```
Comment (hendjoshsr71) [inline libraries/AP_Avoidance/AP_Avoidance.cpp:599]:
```suggestion
        int32_t(packet.alt * 0.1),  // mm -> cm
```

File: libraries/AP_Avoidance/AP_Avoidance.cpp [hunk 2]
```diff
@@ -593,14 +593,17 @@ void AP_Avoidance::handle_msg(const mavlink_message_t &msg)
...
+        Location::AltFrame::ABSOLUTE
+    };
+    const Vector3f vel {
+        packet.vx * 0.01f, // cm to m
+        packet.vy * 0.01f,
+        packet.vz * 0.01f
```
Comment (hendjoshsr71) [inline libraries/AP_Avoidance/AP_Avoidance.cpp:605]:
```suggestion
    const Vector3f vel {
        packet.vx * 0.01, // cm to m
        packet.vy * 0.01,
        packet.vz * 0.01
```

**Review by hendjoshsr71 — CHANGES_REQUESTED** [review]

File: libraries/AP_Avoidance/AP_Avoidance.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Avoidance/AP_Avoidance.cpp]: Oops.  Fixed.

File: libraries/AP_Avoidance/AP_Avoidance.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_Avoidance/AP_Avoidance.cpp]: I've removed these - but I think that's going to cause a bunch of narrowing warnings - so I'll probably re-add them in that case.

**Comment by peterbarker** [comment]
@hendjoshsr71 

```
[809/888] Compiling libraries/AP_Avoidance/AP_Avoidance.cpp
../../libraries/AP_Avoidance/AP_Avoidance.cpp: In member function ‘void AP_Avoidance::handle_msg(const mavlink_message_t&)’:
../../libraries/AP_Avoidance/AP_Avoidance.cpp:603:19: error: narrowing conversion of ‘((double)(int)packet.__mavlink_global_position_int_t::vx * 1.0e-2)’ from ‘double’ to ‘float’ [-Werror=narrowing]
  603 |         packet.vx * 0.01, // cm to m
```

File: libraries/AP_Avoidance/AP_Avoidance.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_Avoidance/AP_Avoidance.cpp:605]: Re-added them.

**Review by hendjoshsr71 — APPROVED** [review]
LGTM

**Comment by hendjoshsr71** [comment]
> @hendjoshsr71
> 
> ```
> [809/888] Compiling libraries/AP_Avoidance/AP_Avoidance.cpp
> ../../libraries/AP_Avoidance/AP_Avoidance.cpp: In member function ‘void AP_Avoidance::handle_msg(const mavlink_message_t&)’:
> ../../libraries/AP_Avoidance/AP_Avoidance.cpp:603:19: error: narrowing conversion of ‘((double)(int)packet.__mavlink_global_position_int_t::vx * 1.0e-2)’ from ‘double’ to ‘float’ [-Werror=narrowing]
>   603 |         packet.vx * 0.01, // cm to m
> ```

Doh. of course

**Pull request merged**